services/ProjectService.js

```diff
@@ -1,5 +1,189 @@
 export default class ProjectService {
   constructor(db) {
     this.client = db.sequelize;
+    this.Project = db.Project;
+    this.User = db.User;
+    this.ProjectMember = db.ProjectMember;
+    this.Link = db.Link;
+  }
+
+  // CREATE PROJECT
+  async create({ name, description, status = "active", ownerId }) {
+    const project = await this.Project.create({ name, description, status });
+
+    // auto-assign creator as admin
+    await this.ProjectMember.create({
+      projectId: project.id,
+      userId: ownerId,
+      isAdmin: true,
+      canAddLinks: true,
+    });
+
+    return this.getById(project.id);
+  }
+
+  // READ
+  async getById(id) {
+    return this.Project.findByPk(id, {
+      include: [
+        { model: this.ProjectMember, as: "projectMembers" },
+        { model: this.Link, as: "links" },
+      ],
+    });
+  }
+
+  async listActive() {
+    return this.Project.findAll({
+      where: { status: "active" },
+      include: [{ model: this.ProjectMember, as: "projectMembers" }],
+      order: [["updatedAt", "DESC"]],
+    });
+  }
+
+  async listByUser(userId) {
+    return this.Project.findAll({
+      include: [
+        {
+          model: this.ProjectMember,
+          as: "projectMembers",
+          where: { userId },
+        },
+      ],
+      order: [["updatedAt", "DESC"]],
+    });
+  }
+
+  // UPDATE
+  async update({ id, userId, args }) {
+    const project = await this.Project.findOne({
+      where: { id },
+      include: [{ model: this.ProjectMember, as: "projectMembers" }],
+    });
+    if (!project) throw new Error("Project not found.");
+
+    const member = project.projectMembers.find((m) => m.userId === userId);
+    if (!member) throw new Error("You are not a member of this project.");
+    if (!member.isAdmin && !member.permissions?.canEdit)
+      throw new Error("You don't have permission to edit this project.");
+
+    await this.Project.update(args, { where: { id } });
+    return this.getById(id);
+  }
+
+  // DELETE (soft delete / archive)
+  async archive({ id, userId }) {
+    const project = await this.Project.findOne({
+      where: { id },
+      include: [{ model: this.ProjectMember, as: "projectMembers" }],
+    });
+    if (!project) throw new Error("Project not found.");
+
+    const member = project.projectMembers.find((m) => m.userId === userId);
+    if (!member?.isAdmin) throw new Error("You don't have permission to archive this project.");
+
+    await this.Project.update({ status: "archived" }, { where: { id } });
+    return this.getById(id);
```

Comment:
Do we need to reload/ `return getById()` here or can we just return the updated instance?
```suggestion
    await project.update({ status: "archived" });
    return project;
```